embedded-graphics/src/lib.rs

```diff
@@ -124,8 +125,8 @@
 //!
 //! fn build_thing(text: &'static str) -> impl Iterator<Item = Pixel<Rgb565>> {
 //!     egrectangle!((0, 0), (40, 40)).into_iter()
-//!         .chain(egcircle!((20, 20), 8, fill_color = Some(Rgb565::RED)))
-//!         .chain(text_6x8!(text, fill_color = Some(Rgb565::GREEN)).translate(Point::new(20, 16)))
+//!         .chain(&egcircle!((20, 20), 8, fill_color = Some(Rgb565::RED)))
+//!         .chain(text_6x8!(text, text_color = Some(Rgb565::GREEN)).translate(Point::new(20, 16)))
```

Comment:
Another option would be to implement the same iter for both `&Thing` and `Thing`. Could use a macro to reduce duplication. Is it even useful to have both?

Reply:
> `egcircle` being a ref and `text_6x8` not being one is odd (and might lead to end user confusion). Not necessarily for this PR, but should we change the text iterators to work with refs so they're consistent with the primitive iters?

Yes this is confusing and wasn't intended to be final, but I tried to keep the changes in the font module to a minimum for this PR. `IntoIterator` is also implemented for `&FontBuilder` so adding a `&` in front of `text_6x8` would already work. I decided not to change this now because I wanted to see the impact of removing `IntoIterator` for `FontBuilder` when working on the font module next.

> Another option would be to implement the same iter for both `&Thing` and `Thing`. Could use a macro to reduce duplication. Is it even useful to have both?

We discussed this earlier in this PR and had decided to only implement `IntoIterator` for `&Thing` for now. Implementing `IntoIterator` for `&Thing` instead of `Thing` has the advantage that the `Thing` object isn't consumed and can be reused multiple times.

Reply:
Ok, that all sounds good!